content/java/concepts/strings/terms/intern/intern.md

```diff
@@ -0,0 +1,42 @@
+---
+Title: '.intern()'
+Description: 'Creates an exact copy of a string located in the heap memory and stores it in the string constant pool.'
+Subjects:
+- 'Computer Science'
+Tags: 
+- 'Java'
+- 'Methods'
+CatalogContent: 
+- 'learn-java'
+- 'paths/computer-science'
+---
+
+The **`.intern()`** method creates an exact copy of a string located in the heap memory and stores it in the string constant pool. With this method, it is possible to optimize memory usage in a Java program by reusing identical string objects.
+
+However, if in the string constant pool exists another string with the same value, a new object won’t be created and the new reference will point to the other string.
+ 
+## Syntax
+
+There are two ways to invoke the intern() method. The first one is used with the new keyword. The second one is directly used on a string literal.
+
+```pseudo
+// Using new keyword
+String <variable name> = new String(<"string value">).intern();
+
+// Used on a string literal
+<String name>.intern();
+```
+
+## Example
+
+```java
+String str1 = "hello";
+String str2 = new String("hello");
+String str3 = str2.intern();
+
+System.out.println(str1 == str2); // false
```

Comment:
We generally don't add the output in the code block as a comment. Please delete the comments and add a shell block. It should like something like:

The output will look like:

```shell
false
true
```